Client/mods/deathmatch/logic/luadefs/CLuaGifDefs.cpp

```diff
@@ -0,0 +1,205 @@
+    gif->Stop();
+    return true;
+}
+
+bool CLuaGifDefs::GifSetProperty(lua_State* luaVM,CClientGif* gif, std::string property, std::variant<int, std::string> value, std::optional<int> frame) {
+    bool valIsNumber = false;
+    int numberValue = -1;
+    if (std::holds_alternative<int>(value)) {
+        valIsNumber = true;
+        numberValue = std::get<int>(value);
+        if (numberValue < 0) {
+            m_pScriptDebugging->LogError(luaVM, "property value can't be a negative number");
+            return false;
+        }
+    }
+    if (property.empty()) {
+        m_pScriptDebugging->LogError(luaVM ,"property can't be empty");
+        return false;
+    }
+    if (frame.has_value()) {
+        int frameCount = gif->GetFrameCount();
+        if (frame > frameCount) {
+            m_pScriptDebugging->LogError(luaVM, "frame doesn't exist");
+            return false;
+        }
+        if (property == "delay") {
+            if (!valIsNumber) {
+                m_pScriptDebugging->LogError(luaVM, "property `delay` must be a number value");
+                return false;
+            }
+            gif->SetFrameDelay(frame.value(), numberValue);
+            return true;
+        }else{
+            m_pScriptDebugging->LogError(luaVM, "property doesn't exist for frame");
+        }
+    }else{
+        if (property == "showing_frame") {
+            if (!valIsNumber) {
+                m_pScriptDebugging->LogError(luaVM, "property `showing_frame` must be a number value");
+                return false;
+            }
+            gif->SetFrame(numberValue);
+            return true;
+        }else{
+            m_pScriptDebugging->LogError(luaVM, "property doens't exist for gif");
+        }
+    }
+    return false;
+}
+
+std::variant<bool, std::string, int> CLuaGifDefs::GifGetProperty(lua_State* luaVM, CClientGif* gif, std::string property, std::optional<int> frame) {
+    if (property.empty()) {
+        m_pScriptDebugging->LogError(luaVM, "property can't be empty");
+        return false;
+    }
+    if (frame.has_value()) {
+        int frameCount = gif->GetFrameCount();
+        if (frame > frameCount) {
+            m_pScriptDebugging->LogError(luaVM, "frame doesn't exist");
+            return false;
+        }
+        if (property == "delay") {
+            return (int)gif->GetFrameDelay(frame.value());
+        }else if (property == "default_delay") {
+            return (int)gif->GetFrameDefaultDelay(frame.value());
+        }else{
+            m_pScriptDebugging->LogError(luaVM, "property doesn't exist for frame");
+        }
+    }else{
+        if (property == "showing_frame") {
+            return (int)gif->GetShowingFrame();
+        }else if (property == "frame_count") {
+            return (int)gif->GetFrameCount();
+        }else if (property == "format") {
+            return (std::string)gif->GetFormat();
+        }else if (property == "tick") {
+            return (int)gif->GetTick();
+        }else{
```

Comment:
Again, need the clang formatting running for all of the changed files in this PR.

Reply:
fixed!